### PR TITLE
Update auto-retry docs for version 2.0 of the plugin

### PR DESCRIPTION
**Pull request by KnorpelSenf**
Version 2.0 is not released yet. When this PQ is ready to be merged, we first need to release the plugin and then we can merge it.

Migrating to 2.0 will be absolutely trivial, since only a single option was renamed. A short note will be included in the news channel post. There is no need for a migration guide.

> Note that in contrast to this, comprehensive migration guides will be published if the core library will receive a major version upgrade.

### CODE REVIEW

**Comment by KnorpelSenf** [comment]
@roj1512 ping

File: site/docs/plugins/auto-retry.md [hunk 1]
```diff
@@ -59,11 +67,30 @@ autoRetry({
...
+  rethrowHttpErrors: true, // do not handle networking errors
+});
+```
 
 ## Plugin Summary
 
 - Name: `auto-retry`
 - Source: <https://github.com/grammyjs/auto-retry>
-- Reference: <https://doc.deno.land/https://raw.githubusercontent.com/grammyjs/auto-retry/main/src/index.ts>
+- Reference: <https://deno.land/x/grammy_auto_retry/mod.ts>
```
Comment (roj1512) [inline site/docs/plugins/auto-retry.md:96]: Should this link to the new ref?

Comment (KnorpelSenf) [reply site/docs/plugins/auto-retry.md:96]: Oops. Yes.

Comment (KnorpelSenf) [reply site/docs/plugins/auto-retry.md]: Fixed

**Review by roj1512 — APPROVED** [review]

**Review by LWJerri — CHANGES_REQUESTED** [review]
Resolve all comments from @deptyped and I will approve the PQ.

**Review by LWJerri — APPROVED** [review]

File: site/docs/plugins/auto-retry.md [hunk 2]
```diff
@@ -5,15 +5,19 @@ next: false
...
 More specifically, this plugin will automatically detect if an API requests fails with a `retry_after` value, i.e. because of rate limiting.
 It will then catch the error, wait the specified period of time, and then retry the request.
 
-::: tip Flood Control
-Telegram will let you know if you send messages too fast.
-This is an important measure for flood control, as it makes sure that your bot does not put Telegram under too much load.
-Using this plugin is important because if you forget to respect [429 errors](../resources/faq#_429-too-many-requests-retry-after-x), Telegram may ban your bot.
-:::
+In addition to handling flood limits, this plugin will retry a request if it fails with an internal server error, i.e. errors with code 500 or larger.
+Networking errors (those that [throw an `HttpError`](../guide/errors.md#the-httperror-object) in grammY) will cause a retry, too.
```
Comment (quadratz) [inline site/docs/plugins/auto-retry.md:16]:
```suggestion
Networking errors (those that [throw an `HttpError`](../guide/errors#the-httperror-object) in grammY) will cause a retry, too.
```

File: site/docs/plugins/auto-retry.md [hunk 3]
```diff
@@ -59,8 +67,27 @@ autoRetry({
...
+```ts
+autoRetry({
+  rethrowInternalServerErrors: true, // do not handle internal server errors
+});
+```
+
+### Rethrowing Networking Errros
+
+You can use `rethrowHttpErrors` to opt out of handling networking errors as described [above](#retry-api-requests-auto-retry).
+If enabled, the thrown [`HttpError`](../guide/errors.md#the-httperror-object) instances are passed on, failing the request.
```
Comment (quadratz) [inline site/docs/plugins/auto-retry.md:84]:
```suggestion
If enabled, the thrown [`HttpError`](../guide/errors#the-httperror-object) instances are passed on, failing the request.
```

File: site/docs/uk/plugins/auto-retry.md [hunk 4]
```diff
@@ -5,15 +5,19 @@ next: false
...
 Зокрема, цей плагін автоматично виявляє, якщо запит API завершується невдало зі значенням `retry_after`, тобто був обмеженний.
 Він перехоплює помилку, чекає вказаний проміжок часу, а потім повторює запит.
 
-::: tip Контролювання перевищення кількості запитів до API
-Telegram повідомить вас, якщо ви надсилаєте повідомлення занадто швидко.
-Це важливий захід для боротьби з перевищенням кількості запитів до API, оскільки він гарантує, що ваш бот не створює надмірного навантаження на Telegram.
-Використання цього плагіна важливе, тому що Telegram може забанити вашого бота, якщо ви забудете враховувати [помилку з кодом статусу 429](../resources/faq#_429-too-many-requests-retry-after-x).
-:::
+Окрім обробки лімітів перевищення кількості запитів, цей плагін повторно виконає запит, якщо він завершився через внутрішню помилку сервера, тобто помилкою з кодом 500 або більшим.
+Мережеві помилки (ті, що [викидають `HttpError`](../guide/errors.md#обʼєкт-httperror) в grammY) також призведуть до повторної спроби.
```
Comment (quadratz) [inline site/docs/uk/plugins/auto-retry.md:16]:
```suggestion
Мережеві помилки (ті, що [викидають `HttpError`](../guide/errors#обʼєкт-httperror) в grammY) також призведуть до повторної спроби.
```

File: site/docs/uk/plugins/auto-retry.md [hunk 5]
```diff
@@ -62,6 +68,28 @@ autoRetry({
...
+```ts
+autoRetry({
+  rethrowInternalServerErrors: true, // не обробляти внутрішні помилки сервера
+});
+```
+
+### Перекидання мережевих помилок
+
+Ви можете використовувати `rethrowHttpErrors`, щоб відмовитися від обробки мережевих помилок, як описано [вище](#повтор-запитів-до-api-auto-retry).
+Якщо цей параметр увімкнено, екземпляри [`HttpError`](../guide/errors.md#обʼєкт-httperror) передаватимуться далі, якщо запит не вдасться виконати.
```
Comment (quadratz) [inline site/docs/uk/plugins/auto-retry.md:85]:
```suggestion
Якщо цей параметр увімкнено, екземпляри [`HttpError`](../guide/errors#обʼєкт-httperror) передаватимуться далі, якщо запит не вдасться виконати.
```

**Review by quadratz — CHANGES_REQUESTED** [review]
@roj1512

**Review by dzikrl — CHANGES_REQUESTED** [review]

File: site/docs/id/plugins/auto-retry.md [hunk 6]
```diff
@@ -5,17 +5,20 @@ next: false
 
 # Pengulang Request API (`auto-retry`)
 
-Plugin ini adalah sebuah [API transformer function](../advanced/transformers).
-Artinya, ia dapat mencegat dan memodifikasi HTTP request yang keluar secara langsung.
+Plugin auto-retry menyediakan sumber daya yang diperlukan untuk menangani [pembatasan flood (flood limits)](../advanced/flood), yaitu sebuah kesalahan dengan kode 429.
+Ia bisa diterapkan ke semua bot selama pengoperasian normal, dan akan sangat membantu terutama selama[proses penyiaran (broadcasting)](../advanced/flood#cara-menyebarkan-pesan) berlangsung.
```
[the remark this answers is not in the document]
Comment (quadratz) [reply site/docs/id/plugins/auto-retry.md:9]: Also simplify a bit?
```suggestion
Ia bisa diterapkan ke semua bot selama pengoperasian normal, terutama saat [proses penyiaran (broadcasting)](../advanced/flood#cara-menyebarkan-pesan) berlangsung.
```

**Review by quadratz — APPROVED** [review]

**Comment by KnorpelSenf** [comment]
I will do a release on Monday or Tuesday

**Pull request merged**